pkg/chunked/storage_linux.go

```diff
@@ -232,28 +241,30 @@ func convertTarToZstdChunked(destDirectory string, blobSize int64, iss ImageSour
 	chunked, err := compressor.ZstdCompressor(f, newAnnotations, &level)
 	if err != nil {
 		f.Close()
-		return nil, "", nil, err
+		return nil, "", "", nil, err
 	}
 
-	digester := digest.Canonical.Digester()
-	hash := digester.Hash()
-
-	if _, err := io.Copy(io.MultiWriter(chunked, hash), diff); err != nil {
+	digesterUncompressed := digest.Canonical.Digester()
```

Comment:
AFAICS now we have `digesterCompressed` (computing a digest of the pre-conversion layer) and `digesterUncompressed` (computing a digest of the converted chunked data`. The naming suggest they correspond to each other, which confuses me a lot. Please use distinct names (`originalRawDigester` / `convertedOutputDigester` or something).